modules/pubsub/pBase.go

```diff
@@ -61,9 +63,11 @@ func (p *PubBase) Subscribe(subscriber IPubSubscriber) bool {
 }
 
 func (p *PubBase) Write(o interface{}) {
+	p.RLock()
 	for i := range p.Subscribers {
 		p.Subscribers[i].write(o)
 	}
+	p.RUnlock()
```

Comment:
I understand you put a _read_ lock because from the Subscribers slice point of view you are only reading, but that means potentially that `Subscribers[i].write()` can be called concurrently for the same Subscriber, and because it's a write operation, could that cause a problem? (i.e is `Subscribers[i].write()` thread-safe?)

Reply:
The thread-safety of the subscriber's write function is variable and not the responsibility of the publisher. This gives the flexibility to implement different subscribers and there are examples of both:
* the SubChannel is based on golang channels and channels are already thread-safe, so the subscriber doesn't need its own locks
* the SubConfig is based on a map and has its own lock